### PR TITLE
Add unit tests for IPv6 Sockets and Code coverage

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
Implement unit tests for sockets IPv6 and 100% code coverage for unit tests.

Test Steps
-----------
Unit tests build and executed. LCOV coverage tested.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_IPv6_Sockets.c [hunk 1]
```diff
@@ -371,14 +371,12 @@ static BaseType_t prv_ntop6_write_short( char * pcDestination,
...
 
-            if( uxLength <= 0U )
-            {
-                xReturn = pdFAIL;
-            }
-            else
-            {
-                pxSet->uxTargetIndex += uxLength;
-            }
+            /* uxLength will be non zero and positive always. */
```
Comment (moninom1) [inline source/FreeRTOS_IPv6_Sockets.c:374]: Good find

File: test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_stubs.c [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        0x00, 0x00,                          /* usIdentification. */
+        0x00, 0x00,                          /* usFragmentOffset. */
+        ipconfigUDP_TIME_TO_LIVE,            /* ucTimeToLive */
+        ipPROTOCOL_UDP,                      /* ucProtocol. */
+        0x00, 0x00,                          /* usHeaderChecksum. */
+        0x00, 0x00, 0x00, 0x00               /* Source IP address. */
+    }
+};
+
+void vPortEnterCritical( void )
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_stubs.c:68]: I do not think it is needed, please check and remove

File: test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_stubs.c [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+ * @brief Convert an ASCII character to its corresponding hexadecimal value.
+ *        Accepted characters are 0-9, a-f, and A-F.
+ *
+ * @param[in] cChar The character to be converted.
+ *
+ * @return The hexadecimal value, between 0 and 15.
+ *         When the character is not valid, socketINVALID_HEX_CHAR will be returned.
+ */
+
+uint8_t ucASCIIToHex( char cChar )
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_stubs.c:85]: Not sure why is it added, Not used can be removed.

File: test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_stubs.c [hunk 4]
```diff
@@ -0,0 +1,117 @@
+/*
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_stubs.c:1]: If you are not using anything from the file, please remove it.

File: test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_utest.c [hunk 5]
```diff
@@ -0,0 +1,1096 @@
...
+ * In FreeRTOS_TCP_IP.c, there is a local macro called 'tcpREDUCED_MSS_THROUGH_INTERNET'.
+ * When a TCP connection is made outside the local network, the MSS
+ * will be reduced to 'tcpREDUCED_MSS_THROUGH_INTERNET' before the connection
+ * is made.
+ */
+#ifndef ipconfigTCP_MSS
+    #define ipconfigTCP_MSS    ( ipconfigNETWORK_MTU - ( ipSIZE_OF_IPv4_HEADER + ipSIZE_OF_TCP_HEADER ) )
+#endif
+
+static const IPv6_Address_t xSampleAddress_IPv6 = { .ucBytes = { 0xfe, 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x70, 0x08 } };
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_utest.c:79]: = { .ucBytes can be removed. xSampleAddress_IPv6 =  { }  should work

File: test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_utest.c [hunk 6]
```diff
@@ -0,0 +1,1096 @@
...
+static const IPv6_Address_t xSampleAddress_IPv6_2 = { .ucBytes = { 0xfe, 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x74, 0x08 } };
+static const IPv6_Address_t xSampleAddress_IPv6_3 = { .ucBytes = { 0xfe, 0x80, 0, 0, 0, 0xde, 0, 0, 0, 0, 0, 0, 0, 0, 0x70, 0x08 } };
+static const IPv6_Address_t xSampleAddress_IPv6_4 = { .ucBytes = { 0xfe, 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0xff, 0, 0, 0, 0x74, 0x08 } };
+static const IPv6_Address_t xSampleAddress_IPv6_5 = { .ucBytes = { 0xfe, 0x80, 0, 0xde, 0, 0xde, 0, 0xde, 0, 0xde, 0xff, 0, 0xde, 0, 0x74, 0x08 } };
+static const IPv6_Address_t xSampleAddress_IPv6_6 = { .ucBytes = { 0xfe, 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 } };
+
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_utest.c:85]: Separation between test and macro can be done to make it more clear.

File: test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_utest.c [hunk 7]
```diff
@@ -0,0 +1,1096 @@
...
+/* IPv6 address pointer passed but socket is not an IPv6 socket */
+void test_pxTCPSocketLookup_IPv6_NotIPv6Socket_NonNULLIPv6Address( void )
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_utest.c:87]: Please add comment with @brief

File: test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_utest.c [hunk 8]
```diff
@@ -0,0 +1,1096 @@
...
+{
+    FreeRTOS_Socket_t xSocket;
+    IPv6_Address_t xAddress_IPv6;
+    FreeRTOS_Socket_t * pxRetSocket;
+
+    xSocket.bits.bIsIPv6 = pdFALSE_UNSIGNED;
+
+    pxRetSocket = pxTCPSocketLookup_IPv6( &xSocket, &xAddress_IPv6, 0xABCD1234 );
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_utest.c:95]: 0xABCD1234 can be globally defined and used everywhere.

File: test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_stubs.c (hunk 3 above)
Comment (tony-josi-aws) [reply test/unit-test/FreeRTOS_Sockets_IPv6/FreeRTOS_Sockets_IPv6_stubs.c:85]: Its used as a stub to make the testing easier, the source files use it.

**Review by moninom1 — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**